src/laser_proc_node.cpp

```diff
@@ -27,24 +27,22 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-/* 
+/*
  * Author: Chad Rockey
  */
 
-#include <laser_proc/LaserTransport.h>
+#include "laser_proc/laser_proc_ros.hpp"
 
-namespace laser_proc {
+#include "rclcpp/rclcpp.hpp"
 
-LaserPublisher LaserTransport::advertiseLaser(ros::NodeHandle& nh, uint32_t queue_size, bool latch){
-  return advertiseLaser(nh, queue_size, ros::SubscriberStatusCallback(), ros::SubscriberStatusCallback(), ros::VoidPtr(), latch);
-}
+int main(int argc, char ** argv)
+{
+  rclcpp::init(argc, argv);
+  auto n = rclcpp::Node::make_shared("laser_proc");
 
-LaserPublisher LaserTransport::advertiseLaser(ros::NodeHandle& nh, uint32_t queue_size,
-                                              const ros::SubscriberStatusCallback& connect_cb,
-                                              const ros::SubscriberStatusCallback& disconnect_cb,
-                                              const ros::VoidPtr& tracked_object, bool latch, bool publish_echoes){
-  return LaserPublisher(nh, queue_size, connect_cb, disconnect_cb, tracked_object, latch, publish_echoes);
-}
+  laser_proc::LaserProcROS lp(n);
 
-} //namespace laser_proc
+  rclcpp::spin(n);
 
+  return 0;
```

Comment:
It's kind of superfluous, but we typically put an `rclcpp::shutdown()` at the end of main.